kmath-ast/src/jvmMain/kotlin/scientifik/kmath/asm/Builders.kt

```diff
@@ -0,0 +1,53 @@
+package scientifik.kmath.asm
+
+import scientifik.kmath.ast.MST
+import scientifik.kmath.ast.evaluate
+import scientifik.kmath.expressions.Expression
+import scientifik.kmath.operations.*
+
+@PublishedApi
+internal fun buildName(expression: AsmExpression<*>, collision: Int = 0): String {
+    val name = "scientifik.kmath.expressions.generated.AsmCompiledExpression_${expression.hashCode()}_$collision"
+
+    try {
+        Class.forName(name)
+    } catch (ignored: ClassNotFoundException) {
+        return name
+    }
+
+    return buildName(expression, collision + 1)
+}
+
+inline fun <reified T> AsmExpression<T>.compile(algebra: Algebra<T>): Expression<T> {
+    val ctx = AsmGenerationContext(T::class.java, algebra, buildName(this))
+    invoke(ctx)
```

Comment:
This method should never be made public, the whole compilation chain: creation of the context, invokation of expression and generation should hidden from API in order to avoid leaking of unfinilized AsmGenerationContext.

Reply:
OK